### PR TITLE
[148] feat delete a landscape if no associated sites. (edit base to develop before merge)

### LINKED ISSUE #148: Create/edit/delete landscape
Details for delete:
- if any sites associated with, show message, disable delete button
- if no sites associated with, dont show message, enable delete button
- show confirmation

### PR DESCRIPTION
**Pull request by mmnoo**
closes #148 

delete a landscape

Details:
- if there are associated sites, some explainer test shows, and the button is disabled
- if there are no associated sites, some different text shows, and the button is enabled
- a generic button was made to show what ever 'holding'/progress text while an operation is in progress.

### CODE REVIEW

**Comment by julia-rom** [comment]
weird this one doesn't have a review app

**Comment by julia-rom** [comment]
I think we should have a big warning for any landscape/site/org deletion since it is a permanent action. Maybe it is a model that pops up and says ' Are you sure? ' if so, please type out the name of the landscape.

@alanjleonard thoughts?

File: mrtt-ui/src/views/LandscapeForm.js [hunk 1]
```diff
@@ -135,6 +153,22 @@ const LandscapeForm = ({ isNewLandscape }) => {
...
     navigate('/landscapes')
   }
+
+  const handleDeleteClick = () => {
+    setIsDeleting(true)
+    axios
+      .delete(landscapeUrl)
+      .then(() => {
+        setIsDeleting(false)
+        toast.success(language.success.getDeleteThingSuccessMessage(landscapeName))
```
Comment (julia-rom) [inline mrtt-ui/src/views/LandscapeForm.js:163]: what thing are we referring to ? can it be more specific?

**Comment by mmnoo** [comment]
> I think we should have a big warning for any landscape/site/org deletion since it is a permanent action. Maybe it is a model that pops up and says ' Are you sure? ' and ' if so, please type out the name of the landscape.'
> 
> @alanjleonard thoughts?

OOOh, ya. I made a to do for that for myself and then forgot to do it. Ill make a new ticket for that and do it right away. (assuming Al is a yes on that from Mermaid and otehr work )

File: mrtt-ui/src/views/LandscapeForm.js (hunk 1 above)
Comment (mmnoo) [reply mrtt-ui/src/views/LandscapeForm.js:163]: Im just reusing the same function to return strings

```
 getEditThingSuccessMessage: (thing) => `${thing} has been edited.`,
  getCreateThingSuccessMessage: (thing) => `${thing}, has been created.`,
  getDeleteThingSuccessMessage: (thing) => `${thing} has been deleted.`
  ```

Comment (mmnoo) [reply mrtt-ui/src/views/LandscapeForm.js:163]: Rather than just having a 500 functions. For timelines.

Comment (julia-rom) [reply mrtt-ui/src/views/LandscapeForm.js:163]: ah ok got it!

**Pull request merged**